api/grpcserver/grpc.go

```diff
@@ -58,28 +58,24 @@ func streamingGrpcLogStart(
 	return handler(srv, stream)
 }
 
-// NewPublic creates a new Server listening on the PublicListener address with the given logger and config.
+// NewWithServices creates a new Server listening on the provided address with the given logger and config.
 // Services passed in the svc slice are registered with the server.
-func NewPublic(logger *zap.Logger, config Config, svc []ServiceAPI) (*Server, error) {
+func NewWithServices(listener string, logger *zap.Logger, config Config, svc []ServiceAPI) (*Server, error) {
 	if len(svc) == 0 {
 		return nil, errors.New("no services to register")
 	}
 
-	server := New(config.PublicListener, logger, config)
-	for _, s := range svc {
-		s.RegisterService(server.GrpcServer)
+	// check if listener IP is in private network range
+	host, _, err := net.SplitHostPort(listener)
+	if err != nil {
+		return nil, fmt.Errorf("split local listener: %w", err)
 	}
-	return server, nil
-}
-
-// NewPrivate creates new Server listening on the PrivateListener address with the given logger and config.
-// Services passed in the svc slice are registered with the server.
-func NewPrivate(logger *zap.Logger, config Config, svc []ServiceAPI) (*Server, error) {
-	if len(svc) == 0 {
-		return nil, errors.New("no services to register")
+	ip := net.ParseIP(host)
+	if !ip.IsPrivate() && !ip.IsLoopback() {
+		logger.Warn("unsecured grpc server is listening on a public IP address", zap.String("address", listener))
 	}
```

Comment:
I'm not sure I understand; `net.ParseIP` can only parse IPs, so it would fail when parsing `localhost`. `127.0.0.1` however is considered `IsPrivate()`.

Reply:
That's my point. The warning will be logged if  `host == "localhost"` even though localhost is a loopback.

Reply:
Updated the check